example/Benchmark.cpp

```diff
@@ -0,0 +1,380 @@
+            file << std::setprecision(14);
+            for (const auto& row : matrix) {
+                for (size_t j = 0; j < row.size(); ++j) {
+                    file << row[j];
+                    if (j < row.size() - 1) file << ",";
+                }
+                file << "\n";
+            }
+        }
+
+// == The following saves the rho and T values corresponding to the indices (for plotting purposes) ==
+void write_vector_csv(const std::string& filename, const std::vector<Real>& vec) {
+    std::ofstream file(filename);
+    file << std::setprecision(14);
+    for (const auto& val : vec) {
+        file << val << "\n";
+    }
+}
+
+
+
+// == Main loop ===
+int main(int argc, char* argv[]) {
+    if (argc < 5) {
+        std::cerr << "Usage: " << argv[0] << " path/to/output nRho nT sp5_file matid1 matid2 ..."; //Is this good usage? 
+        return 1;
+    }
+    fs::path base_output_path(argv[1]);
+    //if the folder does not exist, try to make one
+    if (!fs::exists(base_output_path)) {
+        if (!fs::create_directories(base_output_path)) {
+            std::cerr << "Failed to create output directory: " << base_output_path << "\n";
+            return 1; }}
+    int nRho = std::stoi(argv[2]);  //test on batch sizes of 144, 269, 512, 2048 or round powers of ten
+    int nT = std::stoi(argv[3]); 
+    std::string sp5file = argv[4];
+
+    std::vector<int> matids;
+    for (int i = 5; i < argc; ++i)
+        matids.push_back(std::atoi(argv[i]));
+
+    // == Iterate through each material ==
+    for (int matid : matids) {
+        const int ntimes = 20; // Perform 20 time trials
+
+        // == Get material metadata ==
+        SesameMetadata meta;
+        eosGetMetadata(matid, meta);
+
+        // == Set up bounds ==
+        Real rhoMin = 1.1 * std::max(meta.rhoMin, 1e-5);
+        Real rhoMax = 0.9 * meta.rhoMax;
+        Real TMin   = 1.1 * std::max(meta.TMin, 1.0);
+        Real TMax   = 0.9 * meta.TMax;
+        Bounds rhoBounds(rhoMin, rhoMax, nRho);
+        Bounds TBounds(TMin, TMax, nT);
+
+        std::vector<Real> rhos(nRho), temps(nT);
+        for (int i = 0; i < nRho; ++i) rhos[i] = rhoBounds.i2lin(i);
+        for (int j = 0; j < nT; ++j) temps[j] = TBounds.i2lin(j);
+
+        // == This will save the values of rho and T corresponding to the indices chosen ==
+        std::string rho_file = (base_output_path / ("rho_axis_" + std::to_string(matid) + "_nRho-" + std::to_string(nRho) + ".csv")).string();
+        std::string temp_file = (base_output_path / ("temp_axis_" + std::to_string(matid) + "_nT-" + std::to_string(nT) + ".csv")).string();
+        write_vector_csv(rho_file, rhos);
+        write_vector_csv(temp_file, temps);
+
+        // === Load EOS Models ===
+        SpinerEOSDependsRhoT eos_rt(sp5file, matid);
+        SpinerEOSDependsRhoSieTransformable<ShiftTransform> eos_rs(sp5file, matid); //eventually make it SpinerEOSDependsRhoSie<NullTransfom>, for example
```

Comment:
Maybe let's keep this as a `NullTransform` for now. You can maybe create a new EOS object with the shift transform but comment it out.